### PR TITLE
[cri-o-1.16] Update ocicni to add "WithContext" methods

### PR DESCRIPTION
**Pull request by mccv1r0**
What type of PR is this?

/kind feature

What this PR does / why we need it:

Update ocicni to add "WithContext" methods.
Update CNI vendor'ed code to support context argument

Which issue(s) this PR fixes:
Special notes for your reviewer:
Does this PR introduce a user-facing change?

None

### CODE REVIEW

File: pkg/ocicni/types.go [hunk 1]
```diff
@@ -100,12 +102,21 @@ type CNIPlugin interface {
...
 	SetUpPod(network PodNetwork) ([]types.Result, error)
 
+	// SetUpPodWithContext is the same as SetUpPod but takes a context
+	SetUpPodWithContext(ctx context.Context, network PodNetwork) ([]types.Result, error)
+
 	// TearDownPod is the method called before a pod's sandbox container will be deleted
 	TearDownPod(network PodNetwork) error
 
+	// TearDownPodWithContext is the same as TearDownPod but takes a context
+	//TearDownPodWithContext(ctx context.Context, network PodNetwork) error
```
Comment (haircommander) [inline pkg/ocicni/types.go:112]: why is this commented out?

File: pkg/ocicni/types.go [hunk 2]
```diff
@@ -100,12 +102,21 @@ type CNIPlugin interface {
...
+
 	// Status is the method called to obtain the ipv4 or ipv6 addresses of the pod sandbox
 	GetPodNetworkStatus(network PodNetwork) ([]types.Result, error)
 
+	// GetPodNetworkStatusWithContext is the same as GetPodNetworkStatus but takes a context
+	//GetPodNetworkStatusWithContext(ctx context.Context, network PodNetwork) ([]types.Result, error)
```
Comment (haircommander) [inline pkg/ocicni/types.go:118]: same here

File: pkg/ocicni/types.go (hunk 1 above)
Comment (mccv1r0) [reply pkg/ocicni/types.go]: Good catch

**Comment by mrunalp** [comment]
@dcbw ptal

**Pull request merged**